compiler/codegen/mli.cpp

```diff
@@ -373,6 +374,52 @@ std::string MLIContext::genMarshalBodyStringC(Type* t, bool out) {
   return gen;
 }
 
+std::string MLIContext::genMarshalBodyStringChapel(Type * t, bool out) {
```

Comment:
I'm torn between duplicating this method for Chapel strings and just adding the `c_ptr(c_char)` sensitive logic to the body of `genMarshalBodyStringC`, and renaming it to something more general.

Reply:
I think just removing the "C" portion from the name would okay, since it would only be used for strings?

Reply:
If it gets used more generally than that, we can rename it again